src/lib/types/stores/store.ts

```diff
@@ -0,0 +1,8 @@
+export interface Store<T, K> {
+    get(key: K): Promise<T>;
+    getAll(): Promise<T[]>;
```

Comment:
I wonder if it would make sense to have a companion `queryAll(query: object): Promise<T[]>` method?

Reply:
Also, would a `insert(key: K, value: T): Promise<T>;` make sense?

Reply:
For now I left it up to the separate stores, and since we're using autogenerated keys in some of the stores, we'd probably need to extend the interface type to be `<Insertable, Key, Selectable>` to reflect the keys that gets added on the database side?

Reply:
so `<I, K, S>`
`insert(value: I): Promise<S>`

Reply:
> I wonder if it would make sense to have a companion `query(q: object): Promise<T[]>` method?

definitely, also tempting to consider typing the query as well?
as well as supporting pagination params?